src/browser/components/MainPage.jsx

```diff
@@ -20,8 +20,9 @@ import TabBar from './TabBar.jsx';
 import HoveringURL from './HoveringURL.jsx';
 import Finder from './Finder.jsx';
 import NewTeamModal from './NewTeamModal.jsx';
+import SettingsModal from './SettingsPage.jsx';
 
-export default class MainPage extends React.Component {
+export default class MainPage extends React.PureComponent {
```

Comment:
`handleBadgeChange` is called on an interval and it also calls `setState` which causes `MainPage` to render on the same interval.
PureComponent engages use of shouldComponentUpdate and prevents unnecessary renders.

Reply:
We may see performance improvements here now that render isn't being called on an interval.